cloud_modules/openstack.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Test use of credentials with cloud module, taken from Ansible docs
+  hosts: localhost
+  connection: local
+  gather_facts: False
+
+  tasks:
+
+    - name: Create a new instance, per example (credential_type=openstack)
+      tags:
+        - openstack
+      os_server:
+           state: present
```

Comment:
One thing worth mentioning - in the GCE playbook I wrote, I explicitly only tested `state: absent`, because I didn't want to have to deal with side effects, or cleanup.  I'm not sure that we need to _extensively_ test VM creation - I _mostly_ care about testing that Tower credential injection works and allows you to communicate with the third party services in question.

Reply:
That would be better, absolutely, and if I had a better read of the documentation, I would have done that in all of these cases. If just switching `state: absent` would do the job, then I would be happy to do that. I would even be happy to use the `{{ state }}`, as long as a default is put in.

Reply:
For the one I wrote for GCE, I just stuck with a `'{{ state }}'` and picked a really long, not-likely-to-be-used VM name.  That way I was able to say "hey, this VM doesn't exist."